MonoGame.Framework/Input/Mouse.cs

```diff
@@ -151,87 +63,18 @@ public static MouseState GetState()
         /// <param name="y">Relative vertical position of the cursor.</param>
         public static void SetPosition(int x, int y)
         {
-            UpdateStatePosition(x, y);
-
-#if (WINDOWS && DIRECTX)
-            // correcting the coordinate system
-            // Only way to set the mouse position !!!
-            var pt = Window.PointToScreen(new System.Drawing.Point(x, y));
-#elif WINDOWS
-            var pt = new System.Drawing.Point(0, 0);
-#endif
-
-#if DESKTOPGL || ANGLE
-            Sdl.Mouse.WarpInWindow(PrimaryWindow.Handle, x, y);
-#elif WINDOWS
-            SetCursorPos(pt.X, pt.Y);
-#elif MONOMAC
-            var mousePt = NSEvent.CurrentMouseLocation;
-            NSScreen currentScreen = null;
-            foreach (var screen in NSScreen.Screens) {
-                if (screen.Frame.Contains(mousePt)) {
-                    currentScreen = screen;
-                    break;
-                }
-            }
-
-            var point = new PointF(x, Window.ClientBounds.Height-y);
-            var windowPt = Window.ConvertPointToView(point, null);
-            var screenPt = Window.Window.ConvertBaseToScreen(windowPt);
-            var flippedPt = new PointF(screenPt.X, currentScreen.Frame.Size.Height-screenPt.Y);
-            flippedPt.Y += currentScreen.Frame.Location.Y;
-            
-            
-            CGSetLocalEventsSuppressionInterval(0.0);
-            CGWarpMouseCursorPosition(flippedPt);
-            CGSetLocalEventsSuppressionInterval(0.25);
-#elif WEB
             PrimaryWindow.MouseState.X = x;
             PrimaryWindow.MouseState.Y = y;
```

Comment:
The reason was immediate update because platformsetposition wouldn't set the state of the mouse so get state might have not registered the new position, but now that I think about it, it should get moved to per platform since some platforms don't support setting mouse position.